crypto/s2n_ecc_evp.c

```diff
@@ -17,6 +17,10 @@
 
 #include <openssl/ecdh.h>
 #include <openssl/evp.h>
+#if defined(OPENSSL_IS_AWSLC)
+#include <openssl/mem.h>
```

Comment:
Should we handle this in AWS-LC?

Reply:
Perhaps there is a more centralised place this header could be included in S2N?

Reply:
I think it would be nice if AWS-LC updated their headers to automatically `include <openssl/mem.h>` so that no changes in users was necessary when migrating from OpenSSL, but don't think it required for this PR to be merged.